packages/client/src/createTimeseriesProperty.ts

```diff
@@ -0,0 +1,171 @@
+      return OntologiesV2.OntologyObjectsV2.getFirstPoint(
+        client,
+        await client.ontologyRid,
+        objectApiName,
+        primaryKey,
+        propertyName,
+      ) as Promise<TimeSeriesPoint<T>>;
+    },
+    async getLastPoint() {
+      return OntologiesV2.OntologyObjectsV2.getLastPoint(
+        client,
+        await client.ontologyRid,
+        objectApiName,
+        primaryKey,
+        propertyName,
+      ) as Promise<TimeSeriesPoint<T>>;
+    },
+    async getAllPoints(query: TimeSeriesQuery) {
+      return getAllTimeSeriesPoints(
+        client,
+        objectApiName,
+        primaryKey,
+        propertyName,
+        query,
+      );
+    },
+
+    asyncIterPoints(query: TimeSeriesQuery) {
+      return iterateTimeSeriesPoints(
+        client,
+        objectApiName,
+        primaryKey,
+        propertyName,
+        query,
+      );
+    },
+  };
+}
+
+async function getAllTimeSeriesPoints<T extends string | number>(
+  client: MinimalClient,
+  objectApiName: string,
+  primaryKey: any,
+  propertyName: string,
+  body: TimeSeriesQuery,
+): Promise<Array<TimeSeriesPoint<T>>> {
+  const streamPointsIterator = await OntologiesV2.OntologyObjectsV2
+    .streamPoints(
+      client,
+      await client.ontologyRid,
+      objectApiName,
+      primaryKey,
+      propertyName,
+      { range: getTimeRange(body) },
+    );
+
+  const allPoints: Array<TimeSeriesPoint<T>> = [];
+
+  const reader = streamPointsIterator.stream().getReader();
+  for await (
+    const point of parseStreamedResponse(iterateReadableStream(reader))
+  ) {
+    allPoints.push({
+      time: point.time,
+      value: point.value as T,
+    });
+  }
+  return allPoints;
+}
+
+async function* iterateTimeSeriesPoints<T extends string | number>(
+  client: MinimalClient,
+  objectApiName: string,
+  primaryKey: any,
+  propertyName: string,
+  body: TimeSeriesQuery,
+): AsyncGenerator<TimeSeriesPoint<T>, any, unknown> {
+  const utf8decoder = new TextDecoder("utf-8");
+
+  const streamPointsIterator = await OntologiesV2.OntologyObjectsV2
+    .streamPoints(
+      client,
+      await client.ontologyRid,
+      objectApiName,
+      primaryKey,
+      propertyName,
+      { range: getTimeRange(body) },
+    );
+
+  const reader = streamPointsIterator.stream().getReader();
+  for await (
+    const point of parseStreamedResponse(iterateReadableStream(reader))
+  ) {
+    yield {
+      time: point.time,
+      value: point.value as T,
+    };
+  }
+}
+
+function getTimeRange(body: TimeSeriesQuery): TimeRange {
+  if ("$startTime" in body || "$endTime" in body) {
+    return {
+      type: "absolute",
+      startTime: body.$startTime,
+      endTime: body.$endTime,
+    };
+  }
+  return body.$before
+    ? {
+      type: "relative",
+      startTime: {
+        when: "BEFORE",
+        value: body.$before,
+        unit: TimeseriesDurationMapping[body.$unit],
+      },
+    }
+    : {
+      type: "relative",
+      endTime: {
```

Comment:
It made the most sense to me to set this as an endtime if the query was after, e.g. give me all the high temperatures 2 weeks after today, will give all the temps from today up to 2 weeks later

Reply:
Makes sense. Can we add an invariant check in this function when not in production mode too? That would help anyone using pure JS (or throwing `as any` at their code) as well as serve as an extra validation to our own logic.